src/StreamJsonRpc.Tests/JsonRpcOverloadTests.cs

```diff
@@ -0,0 +1,253 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank;
+using Newtonsoft.Json;
+using StreamJsonRpc;
+using Xunit;
+using Xunit.Abstractions;
+using static JsonRpcMethodAttributeTests;
+
+public class JsonRpcOverloadTests : TestBase
```

Comment:
I would suggest you add only tests that focus on the behavior that you intend to change, or that is at particular risk of regressing with your change that don't have adequate coverage already.
I see some very specialized tests below (e.g. that test race conditions for cancellation) that don't seem at all relevant to your exception filter.

Reply:
Let's be sure to cover sync methods, `async Task` methods as well as `async Task<T>` methods, including cases that throw before and after their first yielding `await`. If the nested `Server` class you define here focuses on those scenarios, and test methods are named after those scenarios, I think it will be much clearer that we're covering the interesting scenarios.

Reply:
Also, I don't see any tests that assert that the streams are actually closed after the fatal exception. That's something of the core of your change, so I believe each test should assert that as well.